<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1001</width>
    <height>679</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>WiLAB ITMO</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout_2">
    <item row="0" column="0">
     <layout class="QVBoxLayout" name="verticalLayout">
      <property name="leftMargin">
       <number>10</number>
      </property>
      <property name="topMargin">
       <number>10</number>
      </property>
      <property name="rightMargin">
       <number>10</number>
      </property>
      <property name="bottomMargin">
       <number>10</number>
      </property>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <widget class="QPushButton" name="button_clear">
          <property name="text">
           <string>Очистить</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer_2">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QLabel" name="connectionLabel">
          <property name="font">
           <font>
            <pointsize>14</pointsize>
           </font>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="alignment">
           <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QTabWidget" name="tabWidget">
        <property name="currentIndex">
         <number>0</number>
        </property>
        <widget class="QWidget" name="tab">
         <attribute name="title">
          <string>По событиям</string>
         </attribute>
         <layout class="QGridLayout" name="gridLayout_3">
          <item row="0" column="0">
           <layout class="QGridLayout" name="gridLayout">
            <property name="leftMargin">
             <number>8</number>
            </property>
            <property name="topMargin">
             <number>8</number>
            </property>
            <property name="rightMargin">
             <number>8</number>
            </property>
            <property name="bottomMargin">
             <number>8</number>
            </property>
            <item row="3" column="0">
             <layout class="QHBoxLayout" name="horizontalLayout_4">
              <item>
               <widget class="QPushButton" name="button_save_s">
                <property name="text">
                 <string>Сохранить</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="label_3">
              <property name="text">
               <string>Независимое срабатывание ворот</string>
              </property>
             </widget>
            </item>
            <item row="2" column="0">
             <widget class="QTableView" name="table_0">
              <property name="autoScroll">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item row="2" column="1">
             <widget class="QTableView" name="table_1"/>
            </item>
            <item row="0" column="0">
             <widget class="QCheckBox" name="on_checkbox">
              <property name="text">
               <string>Работа</string>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QLabel" name="label_5">
              <property name="text">
               <string>Последовательное срабатывание 1 -&gt; 2</string>
              </property>
             </widget>
            </item>
            <item row="3" column="1">
             <layout class="QHBoxLayout" name="horizontalLayout_5">
              <item>
               <widget class="QPushButton" name="button_save_d">
                <property name="text">
                 <string>Сохранить</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item row="6" column="0" colspan="2">
             <widget class="QPushButton" name="button_time_reset">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>Обнулить часы [Backspace]</string>
              </property>
              <property name="shortcut">
               <string>Backspace</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_7">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>В данном режиме можно автоматически сохранять времена прохождения через ворота. Слева в таблице две независимых колонки - время для каждых ворот отдельно. Справа в таблице отмечаются только последовательные прохождения первых, затем вторых ворот. Время в миллисекундах, измеряется относительно запуска программы, либо же его можно обнулять нажатием кнопки &quot;Обнулить&quot; или же клавишей backspace на клавиатуре. Для запуска, поставьте флажок &quot;Работа&quot;. </string>
            </property>
            <property name="wordWrap">
             <bool>true</bool>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="tab_2">
         <attribute name="title">
          <string>По времени</string>
         </attribute>
         <layout class="QGridLayout" name="gridLayout_4">
          <item row="0" column="0">
           <layout class="QVBoxLayout" name="verticalLayout_30">
            <item>
             <layout class="QHBoxLayout" name="layout_intervals">
              <item>
               <layout class="QVBoxLayout" name="verticalLayout_3">
                <item>
                 <layout class="QVBoxLayout" name="verticalLayout_2">
                  <item>
                   <widget class="QLabel" name="label_v0">
                    <property name="font">
                     <font>
                      <pointsize>12</pointsize>
                      <bold>true</bold>
                     </font>
                    </property>
                    <property name="text">
                     <string>Ворота 1</string>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QLabel" name="label">
                    <property name="text">
                     <string>Счетчик импульсов</string>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QLCDNumber" name="value_0_counter">
                    <property name="digitCount">
                     <number>3</number>
                    </property>
                    <property name="segmentStyle">
                     <enum>QLCDNumber::Flat</enum>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QLabel" name="label_2">
                    <property name="text">
                     <string>Длительность импульса (1 и 0)</string>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QLabel" name="value_0_time_up">
                    <property name="font">
                     <font>
                      <pointsize>12</pointsize>
                      <bold>true</bold>
                     </font>
                    </property>
                    <property name="text">
                     <string>----</string>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QLabel" name="value_0_time_down">
                    <property name="font">
                     <font>
                      <pointsize>12</pointsize>
                      <bold>true</bold>
                     </font>
                    </property>
                    <property name="text">
                     <string>----</string>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QLabel" name="label_4">
                    <property name="text">
                     <string>Интервал по фронту, по спаду, с.</string>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QLabel" name="value_0_interval_up_front">
                    <property name="font">
                     <font>
                      <pointsize>12</pointsize>
                      <bold>true</bold>
                     </font>
                    </property>
                    <property name="text">
                     <string>----</string>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QLabel" name="value_0_interval_down_front">
                    <property name="font">
                     <font>
                      <pointsize>12</pointsize>
                      <bold>true</bold>
                     </font>
                    </property>
                    <property name="text">
                     <string>----</string>
                    </property>
                   </widget>
                  </item>
                 </layout>
                </item>
                <item>
                 <layout class="QVBoxLayout" name="verticalLayout_4">
                  <item>
                   <widget class="QLabel" name="label_v1">
                    <property name="font">
                     <font>
                      <pointsize>12</pointsize>
                      <bold>true</bold>
                     </font>
                    </property>
                    <property name="text">
                     <string>Ворота 2</string>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QLabel" name="label">
                    <property name="text">
                     <string>Счетчик импульсов</string>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QLCDNumber" name="value_1_counter">
                    <property name="digitCount">
                     <number>3</number>
                    </property>
                    <property name="segmentStyle">
                     <enum>QLCDNumber::Flat</enum>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QLabel" name="label_2">
                    <property name="text">
                     <string>Длительность импульса (1 и 0)</string>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QLabel" name="value_1_time_up">
                    <property name="font">
                     <font>
                      <pointsize>12</pointsize>
                      <bold>true</bold>
                     </font>
                    </property>
                    <property name="text">
                     <string>----</string>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QLabel" name="value_1_time_down">
                    <property name="font">
                     <font>
                      <pointsize>12</pointsize>
                      <bold>true</bold>
                     </font>
                    </property>
                    <property name="text">
                     <string>----</string>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QLabel" name="label_4">
                    <property name="text">
                     <string>Интервал (по фронту, по спаду)</string>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QLabel" name="value_1_interval_up_front">
                    <property name="font">
                     <font>
                      <pointsize>12</pointsize>
                      <bold>true</bold>
                     </font>
                    </property>
                    <property name="text">
                     <string>----</string>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QLabel" name="value_1_interval_down_front">
                    <property name="font">
                     <font>
                      <pointsize>12</pointsize>
                      <bold>true</bold>
                     </font>
                    </property>
                    <property name="text">
                     <string>----</string>
                    </property>
                   </widget>
                  </item>
                 </layout>
                </item>
               </layout>
              </item>
              <item>
               <widget class="QTableView" name="table_cnts">
                <property name="autoScroll">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
              <item>
               <layout class="QVBoxLayout" name="verticalLayout_7">
                <item>
                 <widget class="QCheckBox" name="autoStore">
                  <property name="text">
                   <string>Автозапись</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QLabel" name="label_6">
                  <property name="text">
                   <string>Частота записи Гц</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QSpinBox" name="timeFreq">
                  <property name="minimum">
                   <number>1</number>
                  </property>
                  <property name="maximum">
                   <number>5</number>
                  </property>
                 </widget>
                </item>
                <item>
                 <spacer name="verticalSpacer_2">
                  <property name="orientation">
                   <enum>Qt::Vertical</enum>
                  </property>
                  <property name="sizeHint" stdset="0">
                   <size>
                    <width>20</width>
                    <height>40</height>
                   </size>
                  </property>
                 </spacer>
                </item>
                <item>
                 <widget class="QPushButton" name="button_add">
                  <property name="sizePolicy">
                   <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                    <horstretch>0</horstretch>
                    <verstretch>0</verstretch>
                   </sizepolicy>
                  </property>
                  <property name="text">
                   <string>Записать [Enter]</string>
                  </property>
                  <property name="shortcut">
                   <string>Return</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QPushButton" name="button_save_t">
                  <property name="text">
                   <string>Сохранить</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <spacer name="verticalSpacer">
                  <property name="orientation">
                   <enum>Qt::Vertical</enum>
                  </property>
                  <property name="sizeHint" stdset="0">
                   <size>
                    <width>20</width>
                    <height>40</height>
                   </size>
                  </property>
                 </spacer>
                </item>
               </layout>
              </item>
             </layout>
            </item>
            <item>
             <widget class="QLabel" name="label_8">
              <property name="text">
               <string>В данном режиме можно вручную или же автоматически сохранять информацию о временнЫх характеристиках сигнала. &quot;Длитиельность импульса&quot; - время от фронта до спада (первое поле), от спада до фронта (второе). &quot;Интервал&quot; - время между последовательными фронтами (по фронту), время между последовательными спадами (по спаду).  Время в миллисекундах. В ручном режиме записать состояние в таблицу можно кнопой &quot;Записать&quot; или нажав Enter. Для активации автоматического логирования, установите частоту (1-5 Гц) и выставьте флажок &quot;Автозапись&quot;</string>
              </property>
              <property name="wordWrap">
               <bool>true</bool>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="tab_3">
         <attribute name="title">
          <string>Тележки</string>
         </attribute>
         <layout class="QGridLayout" name="gridLayout_5">
          <item row="1" column="0">
           <widget class="QTableView" name="table_exp"/>
          </item>
          <item row="0" column="0">
           <layout class="QVBoxLayout" name="verticalLayout_5">
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout">
              <item>
               <widget class="QCheckBox" name="exp_run">
                <property name="text">
                 <string>Запись [Enter]</string>
                </property>
                <property name="iconSize">
                 <size>
                  <width>32</width>
                  <height>32</height>
                 </size>
                </property>
                <property name="shortcut">
                 <string>Return</string>
                </property>
               </widget>
              </item>
              <item>
               <spacer name="horizontalSpacer">
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>40</width>
                  <height>20</height>
                 </size>
                </property>
               </spacer>
              </item>
              <item>
               <widget class="QPushButton" name="exp_save">
                <property name="text">
                 <string>Сохранить</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="label_9">
            <property name="text">
             <string>В данном режиме можно производить не только замеры моментов времени, в которые произошел проход через ворота, но и длительность прохода светонепроницаемой движущейся части через вопрота, что позволяет, зная геомерию объекта, получить его мгновенную скорость. Предполагается возможным, что каждые из двух ворот в рамках одного эксперимента могут сработать от 0 до 2х раз включительно. То есть, первые 4 столбца отвечают за первый проход каждых ворот (Л и П - левые и правые), последние 4 - аналогично для второго прохода. При этом не обязательно, чтобы были все 4 срабатывания. В данном режиме обязательно перед каждым измерением выставлять флажок &quot;Запись&quot;, либо нажимать Enter на клавиатуре. По завершении каждого измерения нужно снять флажок, либо нажать ту же клавишу, только после этого будет доступна новая строка данных. </string>
            </property>
            <property name="wordWrap">
             <bool>true</bool>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
